pytensor/tensor/basic.py

```diff
@@ -1418,6 +1418,8 @@ class Alloc(COp):
     """
 
     _f16_ok = True
+    _output_type_depends_on_input_value = True
```

Comment:
Why isn't it in a some base class with the default False value?

Reply:
A lot of these flags are not in a base class like the `_f16_ok`, but no strong preference

Reply:
Do you prefer that?

Reply:
Added it to `Op`